frontend/app/routes/_gcweb-app.personal-information.preferred-language.edit.tsx

```diff
@@ -67,20 +68,19 @@ export default function PreferredLanguageEdit() {
         {t('personal-information:preferred-language.edit.page-title')}
       </h1>
       <Form method="post">
-        <fieldset className="gc-chckbxrdio">
-          <legend>{t('personal-information:preferred-language.edit.page-title')}</legend>
-          <ul className="list-unstyled lst-spcd-2">
-            {preferredLanguageList?.map((language) => {
-              const radioId = `preferred-language-option-${language.id}`;
-              return (
-                <li key={radioId} className="radio">
-                  <input type="radio" name="preferredLanguage" id={radioId} value={language.id} defaultChecked={userInfo.preferredLanguage === language.id} />
-                  <label htmlFor={radioId}>{getNameByLanguage(i18n.language, language)}</label>
-                </li>
-              );
-            })}
-          </ul>
-        </fieldset>
+        <InputRadios
+          id="preferred-language"
+          name="preferredLanguage"
+          legend={t('personal-information:preferred-language.edit.page-title')}
+          options={
+            preferredLanguageList?.map((language) => ({
+              defaultChecked: userInfo.preferredLanguage === language.id,
+              children: getNameByLanguage(i18n.language, language),
+              value: language.id,
+            })) ?? []
+          }
```

Comment:
Wouldn't it be better to check if there is data in `preferredLanguageList` and then conditionally render the component based on that check?

Reply:
It's "not" recommended by react to return null or empty fragment. It's the responsability of the consument to conditionally render the InputRadios component.

https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null

> In practice, returning null from a component isn’t common because it might surprise a developer trying to render it. More often, you would conditionally include or exclude the component in the parent component’s JSX.

Reply:
But in this case, the code **is in the parent**. The check can't be moved up any further in the component tree. (maybe I'm missing the point of your comment)

As coded, if there's no preferred languages, we will get:

```
Preferred language
{nothing}
```

Wouldn't it make more sense to either
a) not render the entire `<InputRadios>` component, or
b) throw

I don't see any reason to render a label when there are no values to select.

Reply:
I get what you mean. Sorry I thought you were put that logic in `InputRadios` component. I can conditionally render the InputRadios component.

Reply:
Fixed!